target/linux/realtek/files-5.10/Documentation/devicetree/bindings/realtek,rtl8300-timer.yaml

```diff
@@ -0,0 +1,124 @@
+# SPDX-License-Identifier: GPL-2.0
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/timer/realtek,rtl8300-timer.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Realtek Timer Device Tree Bindings
+
+maintainers:
+  - Markus Stockhausen <markus.stockhausen@gmx.de>
+
+description: |
+  The Realtek SOCs of the RTL83XX and RTL93XX series have 5-7 timers. Their
+  speed is derived from the Lexra Bus (LXB) by dividers. Each timer has a block
+  of 4 control registers in the adress range 0xb800xxxx with following start
+  offsets.
+
+  RTL838X: 0x3100, 0x3110, 0x3120, 0x3130, 0x3140
+  RTL839X: 0x3100, 0x3110, 0x3120, 0x3130, 0x3140
+  RTL930X: 0x3200, 0x3210, 0x3220, 0x3230, 0x3240, 0x3250
+  RTL931X: 0x3200, 0x3210, 0x3220, 0x3230, 0x3240, 0x3250, 0x3270
+
+  To make use of the timers the DT needs an address/interrupt definition per CPU
+  core to provide an interrupt driven timer with interval HZ. With an additional
+  reg field one timer is addressed that is taken as a stable 3.125 MHz 
+  clocksource. If the Realtek clock control unit is enabled too the required
+  clocks can be referenced by 'include/dt-bindings/clock/rtl83xx-clk.h'. 
+ 
+properties:
+  compatible:
+    enum:
+      - realtek,rtl8380-timer
+      - realtek,rtl8390-timer
+      - realtek,rtl9300-timer
```

Comment:
add all 'supported' chips, for the simple reason, that device tree's should list their specific ID.

e.g. in the dts I'd expect to see 'realtek,rtl9302, realtek-rtl_otto' so that it can try its specific compatible, and fallback otherwise. Having the 'generic' rtl9300 in the middle makes sense still, e.g. if we need a chip specific quirk, we can, if it's a 'generation' (rtl9300) quirck we can, everything else is handled by otto.

Reply:
In the clock driver we aligned to 0 (instead of x) as last digit. With our current DT structure keep this consistent with the clock driver.